Objective-C/CBLDocument.mm

```diff
@@ -109,107 +225,53 @@ - (BOOL) deleteDocument: (NSError**)outError {
             // Save succeeded; now commit:
-            if (!transaction.commit()) {
+            if (!transaction.commit())
                 return convertError(transaction.error(), outError);
-            }
-            
-            // Reload:
-            if (![self loadDoc_mustExist: NO error: outError])
-                return NO;
-            
-            self.properties = nil;
-            [self resetChangesKeys];
             
+            // Reset:
+            [self setC4Doc: nil];
             return YES;
         }
     }
     return convertError(err, outError);
 }
 
 
-#pragma mark - CBLProperties
-
-
-- (CBLBlob *)blobWithProperties:(NSDictionary *)properties error:(NSError **)error {
-    return [[CBLBlob alloc] initWithDatabase: _database properties:properties error:error];
-}
-
-
-- (BOOL)storeBlob:(CBLBlob *)blob error:(NSError **)error {
-    return [blob installInDatabase: _database error: error];
-}
-
-
-- (void) setHasChanges: (BOOL)hasChanges {
-    if (self.hasChanges != hasChanges) {
-        [super setHasChanges: hasChanges];
-        [_database document: self hasUnsavedChanges: hasChanges];
-    }
-}
-
-
-// Called by CBLProperties superclass after a change is made to the properties.
-- (void) markChangedKey: (NSString*)key {
-    [super markChangedKey: key];
-    [[NSNotificationCenter defaultCenter] postNotificationName: kCBLDocumentChangeNotification
-                                                        object: self];
-}
-
-
-#pragma mark - INTERNAL
-
+#pragma mark - PRIVATE
 
-// Called by the CBLDatabase when the document has changed on disk.
-- (void)changedExternally {
-    // The current API design decision is that when a document has unsaved changes, it should
-    // not update with external changes and should not post notifications. Instead the conflict
-    // resolution will happen when the app saves the document.
 
-    if(!self.hasChanges) {
-        NSError* error;
-        if (![self loadDoc_mustExist: YES error: &error])
-            CBLWarn(Default, @"%@ failed to load external changes: %@", self, error);
-        [self postChangedNotificationExternal:YES];
-    }
+- (BOOL) changed {
+    return _dict.changed;
 }
 
 
-- (void)postChangedNotificationExternal:(BOOL)external {
-    NSDictionary* userInfo = external ? @{kCBLDocumentIsExternalUserInfoKey: @YES} : nil;
-    [[NSNotificationCenter defaultCenter] postNotificationName: kCBLDocumentSavedNotification
-                                                        object: self
-                                                      userInfo: userInfo];
-}
-
-
-#pragma mark - LOADING:
-
-
 // (Re)loads the document from the db, updating _c4doc and other state.
 - (BOOL) loadDoc_mustExist: (BOOL)mustExist error: (NSError**)outError {
     auto doc = [self readC4Doc_mustExist: mustExist error: outError];
     if (!doc)
         return NO;
-    [self setC4Doc: doc];
-    self.hasChanges = NO;
+    [self setC4Doc: [CBLC4Document document: doc]];
```

Comment:
The c4doc is wrapped by CBLC4Document which will free the c4doc when the object is dealloc.